HACKING.rst

```diff
@@ -86,18 +86,129 @@ You can also see all the environments by simply running ``tox list``
 Running ``tox run -m format`` and ``tox run -m lint`` before committing code is
 recommended.
 
-Code Style
-==========
-
 Commits
-#######
+-------
+
+Commit messages are based on the `conventional commit`_ style::
+
+  <type>(<optional scope>): <description>
+
+  <optional body>
+
+  <optional footer>
+
+The commit is divided into three sections: a header, body, and footer.
+
+Header
+======
+
+The header is required and consists of three subsections: a type,
+optional scope, and description. The header must be 72 characters or less.
```

Comment:
Not sure if we want to enforce the length, sometimes it may have to be quite long.

Reply:
Github wraps longer headers, which look like this:

![image](https://github.com/canonical/starbase/assets/60674096/32cfb949-e148-411b-9153-02e09ff3e515)

The other rationale is that the header should be brief and quick to read.

Reply:
Oh, that's looks bad. 😑